cloudplatform/connectivity-apache-httpclient4/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DefaultCsrfTokenRetriever.java

```diff
@@ -88,9 +89,13 @@ private Header retrieveCsrfTokenResponseHeader(
         // Additional headers
         headers.forEach(( k, values ) -> values.forEach(v -> csrfTokenRequest.addHeader(k, v)));
 
-        //The service path gets appended to the destination URI inside the execute request
+        // The service path gets appended to the destination URI inside the execute request
         final HttpResponse csrfResponse = httpClient.execute(csrfTokenRequest);
         final Header header = csrfResponse.getFirstHeader(X_CSRF_TOKEN_HEADER_KEY);
+
+        // Consume the HttpEntity and close the connection
```

Comment:
Maybe:

```suggestion
        // The entity is expected to be null
        // But to be safe, discard any potential payload to ensure the connection is closed
```

To indicate this is not the expected case..

Reply:
I phrased it a bit differently while adding log statement